### PR TITLE
server: Add watch for configmap changes to handle change of options

### PR DESCRIPTION
**Pull request by vatsa287**
**What this PR does / why we need it**:
This PR is extension of #984. It adds ability to handle modifications of volume options passed.
Changes:
- Add process in server pods to keep watch on changes of Configmap, if changed,
  - Regenerate volfiles
  - Send SIGHUP to SHD & Brick processes.

**Which issue(s) this PR fixes**:
Updates: #982 

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: server/watch_volfile_changes.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            shd_pid = out
+    except CommandException as err:
+        logging.error(logf(
+            "Failed to get SHD process PID",
+            error=err
+        ))
+
+    if brick_pid:
+        try:
+            execute("kill", "-HUP", str(brick_pid))
```
Comment (aravindavk) [inline server/watch_volfile_changes.py]: Use `os.kill(brick_pid, signal.SIGHUP)`

File: server/watch_volfile_changes.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+storage_unit_volfile_path = os.path.join(VOLFILES_DIR, "%s.vol" % volfile_id)
+client_volfile_path = os.path.join(VOLFILES_DIR, "%s.vol" % volname)
+shd_volfile_path = os.path.join(VOLFILES_DIR, "glustershd.vol")
+
+data = {}
+info_file_path = os.path.join(VOLINFO_DIR, "%s.info" % volname)
+
+def send_sighup():
+    """ Send SIGHUP to Brick & SHD Process notifing change in volfiles """
```
Comment (aravindavk) [inline server/watch_volfile_changes.py:33]: May be calling shell command easier?

```
out, err = execute("pgrep -f  -a 'glusterfsd|shd')
for line in out.split("\n"):
    pid = line.split(" ", 1)[0]
    os.kill(int(pid), signal.SIGHUP)
```

Log if any errors.

File: server/watch_volfile_changes.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        with open(info_file_path, "r") as info_file:
+            content = info_file.read()
+
+            if last_content is not None and content != last_content:
+                last_content = content
+                logging.info(logf(
+                    "Detected change in configmap content. Regenerating volfiles."
+                ))
+
+                with open(info_file_path) as info_file:
```
Comment (aravindavk) [inline server/watch_volfile_changes.py]: skip this read. Use `content`

File: server/watch_volfile_changes.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+                create_brick_volfile(storage_unit_volfile_path, volname, volume_id, brick_path, data)
+                create_shd_volfile(shd_volfile_path, volname)
+                create_client_volfile(client_volfile_path, data)
+
+                # Send HUP to GlusterFSD & SHD processes
+                send_sighup()
+
+            else:
+                logging.info(logf(
```
Comment (aravindavk) [inline server/watch_volfile_changes.py]: Make this debug. Otherwise, this will flood

Comment (aravindavk) [reply server/watch_volfile_changes.py]: or remove

Comment (vatsa287) [reply server/watch_volfile_changes.py]: Removed

File: server/watch_volfile_changes.py (hunk 3 above)
Comment (vatsa287) [reply server/watch_volfile_changes.py]: done

File: server/watch_volfile_changes.py (hunk 2 above)
Comment (vatsa287) [reply server/watch_volfile_changes.py:33]: Updated

File: server/watch_volfile_changes.py (hunk 1 above)
Comment (vatsa287) [reply server/watch_volfile_changes.py]: Done

File: server/watch_volfile_changes.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        out, _, _ = execute("pgrep", "-f", "-a", "glusterfsd|shd")
+        for line in out.split("\n"):
+            pid = line.split(" ", 1)[0]
+            os.kill(int(pid), signal.SIGHUP)
+    except CommandException as err:
+        logging.error(logf(
+            "Failed to get pid",
+            error=err
+        ))
+    except ProcessLookupError as err:
```
Comment (aravindavk) [inline server/watch_volfile_changes.py]: Move this exception handling inside for loop and include `line` in the log details along with the `error=err`.

Comment (vatsa287) [reply server/watch_volfile_changes.py]: Updated

**Review by aravindavk — APPROVED** [review]

**Pull request merged**